### PR TITLE
wip: fix: Active menu link support for Single Page site

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Highlight current menu item when there is one-page site navigation used
- Added current-menu-item on scroll event
- To create a single page menu the user needs to create Menu as the following image: 

### Screenshots
- https://share.getcloudapp.com/WnuNZEYn

### Types of changes
- Batch run for old users
- Introduced filter for them
- add_filter( 'astra_single_page_active_menu_item_support', '__return_true' );
- JS scroll event

### How has this been tested?
- With the one-page site menu by assigning hash links in a template & accordingly in menu items.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards
- [x] My code has proper inline documentation
- [x] I've included any necessary tests
- [x] I've included developer documentation
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: assets/js/unminified/navigation.js [hunk 1]
```diff
@@ -592,4 +592,75 @@ var astraTriggerEvent = function astraTriggerEvent( el, typeArg ) {
...
+		var primary_header_menu_lists = document.querySelectorAll( '.main-navigation li' );
+
+		// Single page active menu support - Primary Header.
+		if( primary_header_menu_lists.length >= 1 ) {
+
+			var primary_header_href_data = [];
+
+			primary_header_menu_lists.forEach(function(i) {
+
+				var parent_li = i;
```
Comment (premanshup) [inline assets/js/unminified/navigation.js:606]: @imnavanath Can we remove this? Rename `i` to `parent_li` and directly use it?

`primary_header_menu_lists.forEach(function(parent_li) {`


`var href = parent_li.querySelector('.menu-link').getAttribute('href');`

File: assets/js/unminified/navigation.js [hunk 2]
```diff
@@ -592,4 +592,75 @@ var astraTriggerEvent = function astraTriggerEvent( el, typeArg ) {
...
+			var  scrollDistance = window.pageYOffset || document.body.scrollTop;
+
+			// Primary header one-page support.
+			if( primary_header_menu_lists.length >= 1 ) {
+
+				var primary_header_element;
+
+				primary_header_href_data.forEach( function( item ) {
+
+					if( document.querySelector(item) != null ) {
```
Comment (premanshup) [inline assets/js/unminified/navigation.js:626]: @imnavanath Please use strict type checking here if possible.

File: assets/js/unminified/navigation.js [hunk 3]
```diff
@@ -592,4 +592,75 @@ var astraTriggerEvent = function astraTriggerEvent( el, typeArg ) {
...
+
+						var active_element_position = document.querySelector(item).offsetTop;
+
+						active_element_position = active_element_position - 40;
+
+						if( active_element_position - 40 <= scrollDistance ) {
```
Comment (premanshup) [inline assets/js/unminified/navigation.js:632]: @imnavanath Why are we substracting 40 twice?

1. `active_element_position = active_element_position - 40;`
2. `if( active_element_position - 40 <= scrollDistance ) {`

File: assets/js/unminified/navigation.js [hunk 4]
```diff
@@ -592,4 +592,75 @@ var astraTriggerEvent = function astraTriggerEvent( el, typeArg ) {
...
+
+				primary_header_menu_lists.forEach(function(item) {
+
+					item.classList.remove('current-menu-item');
+
+					if( item.querySelector( '.menu-link' ).getAttribute( 'href' ) == primary_header_element ) {
+						item.classList.add( 'current-menu-item' );
+					}
+
+					if( undefined == primary_header_element ) {
```
Comment (premanshup) [inline assets/js/unminified/navigation.js:646]: @imnavanath Please use strict type checking here if possible.

File: assets/js/unminified/navigation.js [hunk 5]
```diff
@@ -592,4 +592,75 @@ var astraTriggerEvent = function astraTriggerEvent( el, typeArg ) {
...
+
+						var href = item.querySelector('.menu-link').getAttribute('href');
+
+						if( ! href.startsWith('#') ) {
+							item.classList.add( 'current-menu-item' );
+						}
```
Comment (premanshup) [inline assets/js/unminified/navigation.js:652]: @imnavanath I think this is causing the multiple menu item active issues - https://a.cl.ly/5zuGd89w

Please check.

File: assets/js/unminified/navigation.js (hunk 1 above)
Comment (imnavanath) [reply assets/js/unminified/navigation.js:606]: Valid one 👍

File: assets/js/unminified/navigation.js (hunk 2 above)
Comment (imnavanath) [reply assets/js/unminified/navigation.js:626]: Valid one 👍

File: assets/js/unminified/navigation.js (hunk 4 above)
Comment (imnavanath) [reply assets/js/unminified/navigation.js:646]: Valid one 👍

File: assets/js/unminified/navigation.js (hunk 3 above)
Comment (imnavanath) [reply assets/js/unminified/navigation.js:632]: Wanted to get a threshold value to decide a menu-item is active. Right now removed the conditional subtraction.

File: assets/js/unminified/navigation.js (hunk 5 above)
Comment (imnavanath) [reply assets/js/unminified/navigation.js:652]: Right, fixed the issue.

**Review by premanshup — APPROVED** [review]

**Pull request closed without merging**